### PR TITLE
Adjust to AdjointFactorization

### PR DESCRIPTION
**Pull request by dkarrasch**
This prepares the ground for https://github.com/JuliaLang/julia/pull/46874 and lets CI run smoothly before and after that PR. I suggest to bump the patch version even though this just changes a test, so that nanosoldier can see this.

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -15,6 +15,12 @@ function testlu(A, MF, BF)
     nothing
 end
 testlu(A::Adjoint, MF::Adjoint, BF) = testlu(parent(A), parent(MF), BF)
+if isdefined(LinearAlgebra, :AdjointFactorization)
+    testlu(A::Adjoint, MF, BF) =
+        testlu(parent(A), parent(MF), BF)
+    testlu(A::Transpose, MF, BF) =
+        testlu(parent(A), parent(MF), BF)
+end
```
Comment (YingboMa) [inline test/runtests.jl:23]:
```suggestion
testlu(A::Union{Transpose, Adjoint}, MF, BF) = testlu(parent(A), parent(MF), BF)
```

Comment (dkarrasch) [reply test/runtests.jl:23]: 🤦 Thanks! It took me a while to figure out what is passed where. Thought it was an easy change (and another... and another), and ended up with a mess. 🤣

**Comment by dkarrasch** [comment]
Shall we merge and release? It works on my PR branch.

**Comment by YingboMa** [comment]
Why do we need to release this change? Only tests are updated.

**Comment by dkarrasch** [comment]
> Why do we need to release this change?

To "verify publicly" that my Julia PR doesn't break the ecosystem via a nanosoldier pkgeval run. There is also another commit after the last release, so it wouldn't be tests only.

**Pull request merged**